include/pinocchio/codegen/cppadcg.hpp

```diff
@@ -113,14 +113,14 @@ namespace pinocchio
     }
   };
 
-  template<typename Scalar>
-  struct ScalarCast<Scalar, CppAD::cg::CG<Scalar>>
-  {
-    static Scalar cast(const CppAD::cg::CG<Scalar> & cg_value)
-    {
-      return cg_value.getValue();
-    }
-  };
+   template <typename NewScalar, typename Scalar>
+   struct ScalarCast<NewScalar, CppAD::cg::CG<Scalar>>
+   {
+     static NewScalar cast(const CppAD::cg::CG<Scalar>& value)
```

Comment:
I don't understand why you changed the name of the variable here. Please make the minimal change.

Reply:
I think the previous casting method is wrong and probably Pinocchio is not still using it. This is because we need to define two `typename` as required in https://github.com/stack-of-tasks/pinocchio/blob/devel/include/pinocchio/fwd.hpp#L104-L111.

Please see more in https://github.com/stack-of-tasks/pinocchio/pull/2608#issuecomment-2706584544.

Reply:
However, if you suggest replacing `NewScalar` with `To` and `Scalar` with `From`, then please let me know. I might have not understood your comment.

Reply:
Reverted change of variable name.